### PR TITLE
Feature/civ 1309 create preview env

### PR DESCRIPTION
**Pull request by LeonardoPalmeiro**
### JIRA link ###

civ 1309

### Change description ###
Create the preview environment with all container 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by laarmada — APPROVED** [review]

**Review by TatsyMaslova — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "test:routes": "jest -c jest.routes.config.js",
     "test:a11y": "jest -c jest.a11y.config.js --maxWorkers 15",
     "sonar-scan": "sonar-scanner",
-    "test:smoke": "jest -c jest.smoketest.config.js",
+    "test:smoke": "jest -c jest.smoketest.config.js --detectOpenHandles",
```
Comment (taomoh) [inline package.json:19]: This shouldn't have been committed

**Review by taomoh — CHANGES_REQUESTED** [review]
I think we need someone with AKS/Kubernetes/Docker expertise review this
Also
`bin/generate-ssl-options.sh` and `bin/init.sh` shouldn't have been modified

File: package.json (hunk 1 above)
Comment (LeonardoPalmeiro) [reply package.json:19]: Yes. you're right!! my  bad..
I fixed.

Comment (LeonardoPalmeiro) [reply package.json:19]: bin/generate-ssl-options.sh and bin/init.sh shouldn't have been modified =>  I didn't change them

Comment (LeonardoPalmeiro) [reply package.json:19]: image

Comment (LeonardoPalmeiro) [reply package.json:19]: image

Comment (taomoh) [reply package.json:19]: > Yes. you're right!! my bad.. I fixed.

It looks like it's still there

Comment (taomoh) [reply package.json]: > bin/generate-ssl-options.sh and bin/init.sh shouldn't have been modified => I didn't change them

That's strange. Why are they showing on the PR then?

Comment (LeonardoPalmeiro) [reply package.json]: it's was because I got some delay in my network
image

Comment (LeonardoPalmeiro) [reply package.json]: bin/generate-ssl-options.sh and bin/init.sh shouldn't have been modified => I didn't change them

That's strange. Why are they showing on the PR then?

I've no idea why this happened.
do you have any idea how to fix that?

Comment (taomoh) [reply package.json]: You changed it. It was a permission change 😃 

https://github.com/hmcts/civil-citizen-ui/commit/90cde5a32ff3f87815d4f9565c60f0ee54c0a982

Comment (LeonardoPalmeiro) [reply package.json]: In this case I fixed a potencial bug.  As far as I know in Linux when we have script without permission of execution will fail.

Comment (LeonardoPalmeiro) [reply package.json]: in this case what I should to do keep as is( with the fixed of a potencial bug) or should I keep as  was?

File: bin/add-roles.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# User used during the CCD import and ccd-role creation
+./bin/utils/ccd-add-role.sh "caseworker-civil"
+./bin/utils/ccd-add-role.sh "caseworker-caa"
+./bin/utils/ccd-add-role.sh "caseworker-approver"
+./bin/utils/ccd-add-role.sh "prd-admin"
+./bin/utils/ccd-add-role.sh "citizen"
```
Comment (timja) [inline bin/add-roles.sh:8]: personally I would do all of these with a loop like you did below

File: package.json (hunk 1 above)
Comment (taomoh) [reply package.json]: You can keep it

File: bin/add-roles.sh (hunk 2 above)
Comment (LeonardoPalmeiro) [reply bin/add-roles.sh]: I agree with you. I changed

File: package.json (hunk 1 above)
Comment (LeonardoPalmeiro) [reply package.json]: Ok thanks

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -1,47 +1,124 @@
...
+}
+
+def checkForEnableHelmLabel(branch_name) {
+    return new GithubAPI(this).getLabelsbyPattern(branch_name, "enableHelm").contains("enableHelm")
+}
 
 withPipeline(type, product, component) {
-  enableAksStagingDeployment()
-  disableLegacyDeployment()
+    pipelineConf = config
```
Comment (timja) [inline Jenkinsfile_CNP:57]: why do you need this?

File: Jenkinsfile_CNP [hunk 4]
```diff
@@ -1,47 +1,124 @@
...
+    disableLegacyDeployment()
+    enableAksStagingDeployment()
+    enableSlackNotifications('#civil_notifications')
```
Comment (timja) [inline Jenkinsfile_CNP:60]: you shouldn't need this, this is only for overriding, it's defaulted at https://github.com/hmcts/cnp-jenkins-config/blob/master/team-config.yml#L561

File: Jenkinsfile_CNP [hunk 5]
```diff
@@ -1,47 +1,124 @@
...
+    }
+    onMaster {
+        loadVaultSecrets(secrets)
+    }
+    onDemo {
+        loadVaultSecrets(secrets)
+    }
+
+    before('smoketest:preview') {
+        sh """
```
Comment (timja) [inline Jenkinsfile_CNP:77]: Better to extract this to a shell script as it's easier to reproduce issues locally than in Jenkins

File: Jenkinsfile_CNP [hunk 6]
```diff
@@ -1,47 +1,124 @@
...
+
+    after('smoketest:preview') {
+        steps.archiveArtifacts allowEmptyArchive: true, artifacts: 'output/**/*'
+    }
 
-  after('build') {
-    yarnBuilder.yarn('build')
+    before('functionalTest:preview') {
+      env.IDAM_API_URL = "https://idam-api.aat.platform.hmcts.net"
+      env.URL="https://xui-civil-citizen-ui-pr-${CHANGE_ID}.service.core-compute-preview.internal"
```
Comment (timja) [inline Jenkinsfile_CNP:99]:
```suggestion
      env.URL = "https://xui-civil-citizen-ui-pr-${CHANGE_ID}.service.core-compute-preview.internal"
```

File: Jenkinsfile_CNP [hunk 7]
```diff
@@ -1,47 +1,124 @@
...
+      env.CIVIL_SERVICE_URL="http://civil-service-civil-citizen-ui-pr-${CHANGE_ID}.service.core-compute-preview.internal"
```
Comment (timja) [inline Jenkinsfile_CNP:100]:
```suggestion
      env.CIVIL_SERVICE_URL = "http://civil-service-civil-citizen-ui-pr-${CHANGE_ID}.service.core-compute-preview.internal"
```

File: bin/add-roles.sh [hunk 8]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+# User used during the CCD import and ccd-role creation
+roles=("caseworker-civil-solicitor" "caseworker-civil-systemupdate" "caseworker-civil-admin" "caseworker-civil-staff" "caseworker-civil-judge" "caseworker-civil" "caseworker-caa" "caseworker-approver" "prd-admin" "citizen")
```
Comment (timja) [inline bin/add-roles.sh:4]: you can probably do one role per line which would be easier to read, or group them together

**Review by timja — APPROVED** [review]
few comments

**Review by taomoh — APPROVED** [review]

**Pull request merged**